src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -61,13 +61,16 @@ def __init__(self, trainer: "pl.Trainer", multiple_trainloader_mode: str = "max_
     def _should_reload_train_dl(self) -> bool:
         """Check if train dataloader should be reloaded."""
         n_epochs = self.trainer.reload_dataloaders_every_n_epochs
-        return n_epochs and (self.trainer.current_epoch - self.trainer._last_train_dl_reload_epoch >= n_epochs)
```

Comment:
`_last_train_dl_reload_epoch` is totally useless since it doesn't do anything special

Reply:
It was used to prevent a reload to happen twice at the same epoch. To guarantee that, for example, calling 

reset_train_dataloader()
reset_train_dataloader()

twice in a row would be a no-op for the second call.

Reply:
if that's the case, will revert this.